tests/NuGetGallery.Facts/Services/PackageMetadataValidationServiceFacts.cs

```diff
@@ -1355,6 +1355,206 @@ private async Task<PackageValidationResult> ValidatePackageWithIcon(string iconP
                     _currentUser);
             }
 
+            public async Task RejectPackagesWithEmbeddedReadme()
+            {
+                _nuGetPackage = GeneratePackageWithUserContent(
+                    readmeFilename: "readme.md",
+                    readmeFileContents: "some readme md");
+
+                var result = await _target.ValidateMetadataBeforeUploadAsync(
+                    _nuGetPackage.Object,
+                    GetPackageMetadata(_nuGetPackage),
+                    _currentUser);
+
+                Assert.Equal(PackageValidationResultType.Invalid, result.Type);
+                Assert.Contains("readme", result.Message.PlainTextMessage);
+            }
+
+            [Fact]
+            public async Task AcceptsPackagesWithEmbeddedReadmeForFlightedUsers()
+            {
+                _nuGetPackage = GeneratePackageWithUserContent(
+                    readmeFilename: "readme.md",
+                    readmeFileContents: "some readme md",
+                    licenseExpression: "MIT",
+                    licenseUrl: new Uri("https://licenses.nuget.org/MIT"));
+                _featureFlagService
+                    .Setup(ffs => ffs.IsUploadEmbeddedReadmeEnabled(_currentUser))
+                    .Returns(true);
+
+                var result = await _target.ValidateMetadataBeforeUploadAsync(
+                    _nuGetPackage.Object,
+                    GetPackageMetadata(_nuGetPackage),
+                    _currentUser);
+
+                Assert.Equal(PackageValidationResultType.Accepted, result.Type);
+                Assert.Null(result.Message);
+                Assert.Empty(result.Warnings);
+            }
+
+            [Theory]
+            [InlineData("<readme><something/></readme>")]
+            [InlineData("<readme><something>readme.md</something></readme>")]
+            public async Task RejectsReadmeElementWithChildren(string ReadmeElement)
```

Comment:
Function argument names should be camerCased.

Reply:
you mean this function argument? we treat Readme as one word.

Reply:
I think Andrei meant this:

```suggestion
            public async Task RejectsReadmeElementWithChildren(string readmeElement)
```